File: frontend/types/evidence.ts
import type { Pagination } from "@/types/contract";

/** backend/app/schemas/evidence.py 그대로. */

export interface EvidenceRequest {
  evidence_request_id: string;
  contract_id: string;
  risk_assessment_id: string | null;
  reason: string;
  evidence_type: string;
  due_date: string | null;
  verification_status: string;
  latest_evidence_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface EvidenceRequestListData {
  items: EvidenceRequest[];
  pagination: Pagination;
}

export interface EvidenceRequestCreate {
  contract_id: string;
  reason: string;
  evidence_type: string;
  risk_assessment_id?: string | null;
  due_date?: string | null;
}

export interface Evidence {
  evidence_id: string;
  evidence_request_id: string;
  file_name: string;
  document_hash: string;
  verification_status: string;
  submitted_at: string;
}

export interface Verification {
  verification_id: string;
  evidence_id: string;
  verification_status: string;
  reviewer_comment: string | null;
  resubmission_required: boolean;
  blockchain_tx_id: string | null;
}

export interface VerificationDecision {
  decision: "approve" | "reject" | "hold";
  reviewer_comment?: string;
}

File: backend/app/schemas/evidence.py
from __future__ import annotations

from datetime import date
from typing import Literal

from pydantic import BaseModel

from app.models.enums import EvidenceType


class EvidenceRequestCreateRequest(BaseModel):
    contract_id: str
    reason: str
    evidence_type: EvidenceType
    risk_assessment_id: str | None = None
    due_date: date | None = None


class EvidenceRequestResponse(BaseModel):
    evidence_request_id: str
    contract_id: str
    risk_assessment_id: str | None = None
    reason: str
    evidence_type: str
    due_date: str | None = None
    verification_status: str
    latest_evidence_id: str | None = None
    created_at: str
    updated_at: str


class EvidenceRequestListResponse(BaseModel):
    items: list[EvidenceRequestResponse]
    pagination: dict


class EvidenceResponse(BaseModel):
    evidence_id: str
    evidence_request_id: str
    file_name: str
    document_hash: str
    verification_status: str
    submitted_at: str


class VerificationResponse(BaseModel):
    verification_id: str
    evidence_id: str
    verification_status: str
    reviewer_comment: str | None = None
    resubmission_required: bool
    blockchain_tx_id: str | None = None


class VerificationDecisionRequest(BaseModel):
    decision: Literal["approve", "reject", "hold"]
    reviewer_comment: str | None = None
